var/spack/repos/builtin/packages/strumpack/package.py

```diff
@@ -71,6 +71,8 @@ class Strumpack(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('zfp', when='+zfp')
     depends_on('hipblas', when='+rocm')
     depends_on('rocsolver', when='+rocm')
+    depends_on('slate', when='+slate+mpi')
+    depends_on('slate+cuda', when='+cuda+slate+mpi')
```

Comment:
```suggestion
    depends_on('slate', when='+slate')
    depends_on('slate+cuda', when='+cuda+slate')
```

I think this is simpler since you already have `conflicts('+slate', when='~mpi')`.